### DISCUSSION TITLE
MI100 ROCm5.6 Half speed of 3090

### DISCUSSION THREAD

**Discussion by ccbadd** [ROCm/ROCm#2382] [Q&A] [answered]
I have recently setup a server with 2 MI100s to compare to anther server I have that is running an NV 3090 and P40.  By the specs the MI100s are 1/2 the speed of the 3090 with FP32 but 4 1/2 X as fast with FP16.  I have been testing some LLMs and all of them run at half the speed of the 3090.  I was under the understanding that most LLMs use FP16 rather than FP32.  Is there something I am missing here?

Thanks.

**Comment by fluidnumericsJoe** [comment]
It's possible the code you are running has not been tuned/optimized for CDNA hardware. Do you have profiles of benchmarks on the two different system ?

**Comment by ccbadd** [comment]
I am just comparing inferencing with exllama and now also llama.cpp since they added rocm support.  The benchmark that I am using is the statistics when running Ooobabooga webui so nothing fancy.

**Comment by fluidnumericsJoe** [comment]
Sounds fancy :) Not everyone who programs GPUs does ML (like myself) so I have no idea what exllama, llama.cpp, or Ooobabooga is.

 Do you have a hotspot profile showing which kernels are taking up the most time ? From here, we'd want to  at least understand basic launch parameters, register usage, and lds usage to understand occupancy. To answer your questions, you could get familiar with using rocprof to generate a profile. If this is out of your comfort zone, this sounds like a question for the developers of "exllama"

Just because one GPU has better theoretical peak performance, it does not mean an application will run better on that GPU.

**Comment by tcgu-amd** [comment, marked as answer]
@ccbadd, as @fluidnumerics-joe have stated, it is likely because those projects are primarily designed for NVIDIA GPUs and are not optimized for M100. It may even default to fp32 mode. I think the best to test performance is to run some simple mixed-precision benchmarks using pytorch and observe the performance gaps between 3090 and MI100.

**Reply by ccbadd** [reply]
> @ccbadd, as @fluidnumerics-joe have stated, it is likely because those projects are primarily designed for NVIDIA GPUs and are not optimized for M100. It may even default to fp32 mode. I think the best to test performance is to run some simple mixed-precision benchmarks using pytorch and observe the performance gaps between 3090 and MI100.

This is a pretty old thread but just to update things.  With the current version of ollama the pair of MI100's are roughly just as fast as the pair of 3090s I have running so things have improved greatly.  They are actually faster if I'm running a model that does not fit on the combined 48GB of vram of the 3090s but does fit on the 64GB of MI100 vram.
